### PR TITLE
Delivers data url to safari Issue 127

### PR DESCRIPTION
**Pull request by claytron5000**
Addressing this issue: https://github.com/react-csv/react-csv/issues/127

### CODE REVIEW

**Comment by claytron5000** [comment]
Updated to reduce calls to isSafari, and the cdn minimized code.

**Comment by kronenbourg** [comment]
When will this commit be released? Would like the download to work for iOS devices

**Comment by claytron5000** [comment]
@kronenbourg I haven't heard from the maintainer. The code change I made has no tests to run against. I see the issue being we're using the global `window` object. I have an idea that part of the module could be refactored to use an injected window object, thus we could test with injected window objects to simulated different browsers. This might be a breaking change, I haven't gone down that path. 

Maybe you can review the code that's present and make any suggestions, confirm current tests pass. I imagine more action on the issue will bring more attention.

File: src/core.js [hunk 1]
```diff
@@ -77,13 +77,13 @@ export const toCSV = (data, headers, separator) => {
...
   const URL = window.URL || window.webkitURL;
-
-  return (typeof URL.createObjectURL === 'undefined')
-    ? dataURI
-    : URL.createObjectURL(blob);
+  if (typeof URL.createObjectURL === 'undefined' || isSafari()) {
+    const dataURI = `data:text/csv;charset=utf-8,${uFEFF ? '\uFEFF' : ''}${csv}`;
+    return dataURI;
+  }
+  else {
```
Comment (Partysun) [inline src/core.js]: @claytron5000 You can remove `else`.

Comment (claytron5000) [reply src/core.js]: Updated, thanks

File: cdn/react-csv-1.0.19.js [hunk 2]
```diff
@@ -359,13 +359,13 @@ var toCSV = exports.toCSV = function toCSV(data, headers, separator) {
...
   var csv = toCSV(data, headers, separator);
-  var type = isSafari() ? 'application/csv' : 'text/csv';
-  var blob = new Blob([uFEFF ? "\uFEFF" : '', csv], { type: type });
-  var dataURI = "data:" + type + ";charset=utf-8," + (uFEFF ? "\uFEFF" : '') + csv;
-
   var URL = window.URL || window.webkitURL;
-
-  return typeof URL.createObjectURL === 'undefined' ? dataURI : URL.createObjectURL(blob);
+  if (typeof URL.createObjectURL === 'undefined' || isSafari()) {
+    var dataURI = "data:text/csv;charset=utf-8," + (uFEFF ? "\uFEFF" : '') + csv;
```
Comment (Partysun) [inline cdn/react-csv-1.0.19.js]: only return "data:text... @claytron5000

File: cdn/react-csv-1.0.19.js [hunk 3]
```diff
@@ -359,13 +359,13 @@ var toCSV = exports.toCSV = function toCSV(data, headers, separator) {
...
+    return dataURI;
+  }
+  var blob = new Blob([uFEFF ? "\uFEFF" : '', csv], 'application/csv');
```
Comment (Partysun) [inline cdn/react-csv-1.0.19.js]: You can use const instead of var @claytron5000

**Comment by claytron5000** [comment]
Declined because there's a much simpler way.

**Comment by Partysun** [comment]
@claytron5000 Did we have safari support in the master branch?

**Comment by wortkotze** [comment]
Hi @claytron5000, Did u find time to double check it.
For me it's really urgent as we need the topic to download personal data out of a profil.
I tried to add _self as proposed to the component but it's still not working on apple iphones :(

any timing behind the PR? thx Stephan

**Pull request closed without merging**